assignments/assignment03.rb

```diff
@@ -48,3 +96,172 @@
 # - WithdrawalTransaction
 
 # use blocks for your HTML rendering code
+
+
+
+
+#The output isn't working properly.  I was happy to get it this close though.
+#It was difficult to work through the homework without having the lecture
+#recordings to reference.
+
+
+class BankAccount
+  attr :records
+  def initialize csvFileData
+    @records = csvFileData.map do |line|
+      fields = line.split ','
+      create_transaction fields[0], fields[1], fields[2], fields[3].chop
+    end
+
+    @records = records.sort_by { |line| (line[:date].split"/")[1].to_i}
+  end
+
+  def create_transaction(date, payee, amount, type)
+    {date: date, payee: payee, amount: amount, type: type}
+  end
+
+
+  # - summary:
+  #   - starting balance, total deposits, total withdrawals, ending balance
+  # summary = [{start_bal, end_bal, total_d, total_w}]
+  # access by typing december.records.summary
+  def summary
+    total_w = 0
+    total_d = 0
+    self.records.each do |line|
+      if line[:type].downcase == "withdrawal"
+        total_w += line[:amount].to_f.round(2)
+      elsif line[:type].downcase == "deposit"
+        total_d += line[:amount].to_f.round(2)
+      end
+    end
+
+    end_bal = total_d - total_w
+    {start_bal: 0.00, end_bal: end_bal, total_d: total_d, total_w: total_w }
+
+  end
+
+  def extract_withdrawal
+    withdrawals = []
+    self.records.each do |line|
```

Comment:
use select on records to generate the collection of withdrawals, instead of using each